### ISSUE TITLE
can this code run in the Windows system？

### ISSUE THREAD

**Issue by improgress** [saic-mdal/lama#148]

**Comment by jackieylogan** [comment]
i think it can, because somebody had put it on pyqt5 version

**Comment by improgress** [comment]
OK thanks a lot




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年9月2日(星期五) 中午12:08
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [saic-mdal/lama] can this code run in the Windows system？ (Issue #148)





 
i think it can, because somebody had put it on pyqt5 version

**Comment by inhahe** [comment]
It doesn't seem to work on Windows. When I entered the command 'conda env create -f conda_env.yml', I got this:

C:\Users\inhah\AppData\Roaming\Python\Python37\site-packages\requests\__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.11) or chardet (5.0.0) doesn't match a supported version!
  RequestsDependencyWarning)
Collecting package metadata (repodata.json): done
Solving environment: failed

ResolvePackageNotFound:
  - jpeg==9b=h024ee3a_2
  - expat==2.4.1=h2531618_2
  - cryptography==3.4.7=py36hd23ed53_0
  - fontconfig==2.13.1=h6c09931_0
  - protobuf==3.17.2=py36h295c915_0
  - lcms2==2.12=h3be6417_0
  - libvpx==1.7.0=h439df22_0
  - scikit-image==0.17.2=py36h284efc9_4
  - libblas==3.9.0=11_linux64_mkl
  - openh264==2.1.0=hd408876_0
  - libgomp==9.3.0=h5101ec6_17
  - pcre==8.45=h295c915_0
  - tk==8.6.11=h1ccaba5_0
  - pillow==8.3.1=py36h2c7a002_0
  - mkl==2021.3.0=h06a4308_520
  - aiohttp==3.7.4.post0=py36h7f8727e_2
  - scikit-learn==0.24.2=py36ha9443f7_0
  - setuptools==58.0.4=py36h06a4308_0
  - _openmp_mutex==4.5=1_gnu
  - ca-certificates==2021.7.5=h06a4308_1
  - qt==5.9.7=h5867ecd_1
  - markdown==3.3.4=py36h06a4308_0
  - brotlipy==0.7.0=py36h27cfd23_1003
  - openssl==1.1.1l=h7f8727e_0
  - pandas==1.1.5=py36h284efc9_0
  - idna_ssl==1.1.0=py36h06a4308_0
  - matplotlib==3.3.4=py36h06a4308_0
  - certifi==2021.5.30=py36h06a4308_0
  - libxml2==2.9.12=h03d6c58_0
  - libopus==1.3.1=h7b6447c_0
  - gmp==6.2.1=h2531618_2
  - blinker==1.4=py36h06a4308_0
  - cudatoolkit==10.2.89=hfd86e86_1
  - bzip2==1.0.8=h7b6447c_0
  - c-ares==1.17.1=h27cfd23_0
  - tabulate==0.8.9=py36h06a4308_0
  - yaml==0.2.5=h7b6447c_0
  - xz==5.2.5=h7b6447c_0
  - libtasn1==4.16.0=h27cfd23_0
  - importlib-metadata==4.8.1=py36h06a4308_0
  - multidict==5.1.0=py36h27cfd23_2
  - cffi==1.14.6=py36h400218f_0
  - chardet==4.0.0=py36h06a4308_1003
  - zlib==1.2.11=h7b6447c_3
  - readline==8.1=h27cfd23_0
  - libunistring==0.9.10=h27cfd23_0
  - scipy==1.5.3=py36h9e8f40b_0
  - pip==21.0.1=py36h06a4308_0
  - libpng==1.6.37=hbc83047_0
  - freetype==2.10.4=h5ab3b9f_0
  - libgcc-ng==9.3.0=h5101ec6_17
  - coverage==5.5=py36h27cfd23_2
  - ninja==1.10.2=hff7bd54_1
  - pysocks==1.7.1=py36h06a4308_0
  - sip==4.19.8=py36hf484d3e_0
  - yarl==1.6.3=py36h27cfd23_0
  - libxcb==1.14=h7b6447c_0
  - libuuid==1.0.3=h1bed415_2
  - lame==3.100=h7b6447c_0
  - openjpeg==2.4.0=h3ad879b_0
  - libwebp-base==1.2.0=h27cfd23_0
  - lz4-c==1.9.3=h295c915_1
  - gst-plugins-base==1.14.0=h8213a91_2
  - grpcio==1.36.1=py36h2157cd5_1
  - absl-py==0.13.0=py36h06a4308_0
  - liblapack==3.9.0=11_linux64_mkl
  - libtiff==4.2.0=h85742a9_0
  - pyjwt==2.1.0=py36h06a4308_0
  - pyyaml==5.4.1=py36h27cfd23_1
  - cytoolz==0.11.0=py36h7b6447c_0
  - libuv==1.40.0=h7b6447c_0
  - libidn2==2.3.2=h7f8727e_0
  - zstd==1.4.9=haebb681_0
  - tifffile==2020.10.1=py36hdd07704_2
  - glib==2.69.1=h5202010_0
  - python==3.6.13=h12debd9_1
  - omegaconf==2.1.1=py36h5fab9bb_0
  - numpy==1.19.5=py36hfc0c790_2
  - libprotobuf==3.17.2=h4ff587b_1
  - libstdcxx-ng==9.3.0=hd4cf53a_17
  - gnutls==3.6.15=he1e5248_0
  - nettle==3.7.3=hbbd107a_1
  - sqlite==3.36.0=hc218d9a_0
  - ncurses==6.2=he6710b0_1
  - libffi==3.3=he6710b0_2
  - intel-openmp==2021.3.0=h06a4308_3350
  - ld_impl_linux-64==2.35.1=h7274673_9
  - icu==58.2=he6710b0_3
  - x264==1!157.20191217=h7b6447c_0
  - libcblas==3.9.0=11_linux64_mkl
  - pyqt==5.9.2=py36h05f1152_2
  - libgfortran5==9.3.0=ha5ec8a7_17
  - gstreamer==1.14.0=h28cd5cc_2
  - dbus==1.13.18=hb2f20db_0
  - matplotlib-base==3.3.4=py36h62a2d02_0
  - cython==0.29.24=py36h295c915_0
  - kiwisolver==1.3.1=py36h2531618_0
  - libgfortran-ng==9.3.0=ha5ec8a7_17
  - pywavelets==1.1.1=py36h7b6447c_2
  - ffmpeg==4.2.2=h20bf706_0
  - async-timeout==3.0.1=py36h06a4308_0

Maybe something else went wrong, I'm not sure.

I didn't try it outside of conda because installing pytorch never works for me outside of conda.

**Comment by shahJaimin77** [comment]
Hi @inhahe , It works on Windows. But as you mentioned creating conda environment using their yml file won't work.
You can create a normal conda environment using:
`conda create -n lama_env python=3.9`

And then installing the following dependencies:
`conda install pytorch torchvision torchaudio cudatoolkit=10.2 -c pytorch -y
pip install pytorch-lightning==1.2.9`

This worked for me!

**Comment by liyiersan** [comment]
Hi, @shahJaimin77 , I met some problems when calculating metrics on windows , did the evaluate_predicts.py work for on Windows?

**Comment by naga-k** [comment]
I think the folks who built the env yml file need to remove the platform-specific build constraints as described in this article.
https://stackoverflow.com/questions/55554431/conda-fails-to-create-environment-from-yml

**Comment by shahJaimin77** [comment]
Yes, if we are using their environment file, then this problem would occur because of the platform-specific build constraints.
But its better we create our own env file and install the packages as mentioned above.

**Comment by shahJaimin77** [comment]
> Hi, @shahJaimin77 , I met some problems when calculating metrics on windows , did the evaluate_predicts.py work for on Windows?

@liyiersan Sorry, but I haven't used the evaluate_predicts.py file

**Comment by naga-k** [comment]
@shahJaimin77 , thanks for replying. how did you mange to install all the packages?

**Comment by shahJaimin77** [comment]
@naga-k 
Install the dependencies using the following commands:
`conda install pytorch torchvision torchaudio cudatoolkit=10.2 -c pytorch -y`
`pip install pytorch-lightning==1.2.9`

**Comment by haaappytoast** [comment]
For anyone who might have trouble running this code, here are the system specs I used.

**I used wsl2 in windows11**

### System Spec
GPU: NVIDIA GeForce RTX 4060 Ti
OS: Windows 11 (WSL2)
CUDA Version: 12.3
CUDNN Version: 8.9.7.29
Driver Version: 560.94
Python Version: 3.12
Tensorflow Version: 1.16.1

after making virtual env with the command the author provided, I manually installed packages with pip
```
# make virtual env
virtualenv inpenv --python=/usr/bin/python3
source inpenv/bin/activate
```

```
# (to use DDPPlugin, install pytorch-lightning<=1.6.0)
# (to use DualIAATransform, install albumentations==0.4.6)

# install packages listed in requirements.txt with pip 
pip install hydra-core==1.3.2 pytorch-lightning==1.5.1 \
albumentation webdataset==0.2.111 scikit-image==0.25.2 scikit-learn \
easydict pandas joblib matplotlib tabulate kornia
```
